data-prepper-plugins/aws-lambda/src/main/java/org/opensearch/dataprepper/plugins/lambda/processor/LambdaProcessor.java

```diff
@@ -281,6 +283,14 @@ public Collection<Record<Event>> doExecute(Collection<Record<Event>> records) {
         if (recordsToLambda.size() == 0) {
             return resultRecords;
         }
+        // Check if streaming is enabled
+        if (lambdaProcessorConfig.getStreamingOptions() != null && 
```

Comment:
It would probably be better to split the code similar to how we do the LambdaHandler methods. For example, a `LambdaInvoker` interface.

Here is a rough snippet:

```
private final LambdaInvoker lambdaInvoker;
```

In the constructor:

```
if (lambdaProcessorConfig.getStreamingOptions() != null && 
   lambdaProcessorConfig.getStreamingOptions().isEnabled() &&
   lambdaProcessorConfig.getInvocationType() == InvocationType.STREAMING_RESPONSE) {
  lambdaInvoker = new StreamingLambdaInvoker(...);
} else {
  lambdaInvoker = new InvokeLambdaInvoker(...);
```

Then, in this code:

```
lambdaInvoker.invoke(...);
```